tests/test_views.py

```diff
@@ -2,26 +2,36 @@
 
 from cms.models import User
 
-from .base import BaseUrlTestCase
+from .base import BaseTestCase
 
 
-class ContentObjectAutoFillTestCases(BaseUrlTestCase):
+class ContentObjectAutoFillTestCases(BaseTestCase):
     def test_select2_view_no_content_id(self):
         with self.login_user_context(self.superuser):
-            with self.assertRaises(ValueError) as err:
-                self.client.get(self.select2_endpoint)
-            self.assertEqual(
-                str(err.exception), "Content type with id None does not exists."
-            )
+            response = self.client.get(self.select2_endpoint)
+            self.assertEqual(response.status_code, 400)
 
-    def test_select2_view_no_permission(self):
+    def test_select2_view_anonymous_user(self):
+        """HTTP get shouldn't allowed for anonymous user"""
         response = self.client.get(self.select2_endpoint)
-        self.assertEqual(response.status_code, 403)
+        expected_url = "/en/admin/login/?next=" + self.select2_endpoint
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, expected_url)
 
     def test_select2_view_endpoint_post(self):
         """HTTP post shouldn't allowed on this endpoint"""
         response = self.client.post(self.select2_endpoint)
-        self.assertEqual(response.status_code, 403)
+        expected_url = "/en/admin/login/?next=" + self.select2_endpoint
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, expected_url)
+
+    def test_select2_view_endpoint_user_with_no_perm(self):
```

Comment:
I think there probably should also be a test for a post request and user with no perms.